core/src/main/scala/kafka/server/DynamicBrokerConfig.scala

```diff
@@ -799,6 +800,9 @@ class DynamicMetricsReporters(brokerId: Int, server: KafkaBroker) extends Reconf
     reporters.forEach { reporter =>
       metrics.addReporter(reporter)
       currentReporters += reporter.getClass.getName -> reporter
+      if (reporter.clientReceiver() != null) {
```

Comment:
`clientReceiver` is invoked here twice. We should either document that the method can be invoked multiple times or else store the result of the initial invocation so that we can just pass that in to the `add` method below.